### PR TITLE
actually set default TLS settings

### PR DESCRIPTION
**Pull request by wdpypere**

### CODE REVIEW

**Comment by wdpypere** [comment]
@wpoely86 this is the PR I was talking about

File: lib/vsc/ldap/utils.py [hunk 1]
```diff
@@ -67,13 +67,13 @@ def __init__(self, url=None, password=None, connection_dn=None, validation_metho
 
         self.log = getLogger(self.__class__.__name__)
 
-    def tls_settings(self, url):
+    def tls_settings(self):
```
Comment (wpoely86) [inline lib/vsc/ldap/utils.py:70]: you need to change L129 too

File: lib/vsc/ldap/utils.py [hunk 2]
```diff
@@ -67,13 +67,13 @@ def __init__(self, url=None, password=None, connection_dn=None, validation_metho
...
         """
         Set up the LDAP for a TLS connection.
         """
-
-        # FIXME:maybe we want some default configuration here?
-        pass
+        ldap.set_option(ldap.OPT_X_TLS_DEMAND, True)
+        ldap.set_option(ldap.OPT_X_TLS_REQUIRE_CERT, True)
+        ldap.set_option(ldap.OPT_X_TLS_NEWCTX, 0)
```
Comment (wpoely86) [inline lib/vsc/ldap/utils.py:76]: what is this?

Comment (wdpypere) [reply lib/vsc/ldap/utils.py:76]: making sure the connection actually starts using TLS settings, e.g., don't use what was set before.

Comment (wdpypere) [reply lib/vsc/ldap/utils.py:76]: otherwise these settings are ignored

File: lib/vsc/ldap/utils.py (hunk 1 above)
Comment (wdpypere) [reply lib/vsc/ldap/utils.py:70]: no. when looking in vsc-ldap-extensions there is a LUMA part which (wrongly) uses this. I don't know if this is used anywhere. but I think it effectively does NOT use TLS, how hard it even tries.

Comment (wdpypere) [reply lib/vsc/ldap/utils.py:70]: actually, If I read the vsc-ldap codes flying around I'll need to keep and use the `url` var. but some of this is blatantly wrong. I'll try to figure out and fix it all.

**Comment by wdpypere** [comment]
It looks like the function is not used anywhere. but anything is better than `pass` I think.

**Pull request merged**